app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt

```diff
@@ -141,10 +156,33 @@ private fun authenticatedCopyNSec(
     scope: CoroutineScope,
     account: Account,
     clipboardManager: ClipboardManager,
+    keyguardLauncher: ManagedActivityResultLauncher<Intent, ActivityResult>
 ) {
     val fragmentContext = context.getFragmentActivity()!!
-    val authenticators = BIOMETRIC_STRONG or DEVICE_CREDENTIAL
+    val keyguardManager =
+        fragmentContext.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+
+    if (!keyguardManager.isDeviceSecure) {
+        copyNSec(context, scope, account, clipboardManager)
+        return
+    }
+
+    fun keyguardPrompt() {
+        val intent = keyguardManager.createConfirmDeviceCredentialIntent(
+            context.getString(R.string.app_name_release),
+            context.getString(R.string.copy_my_secret_key),
+        )
+
+        keyguardLauncher.launch(intent)
+    }
+
+    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q) {
+        keyguardPrompt()
+        return
+    }

```

Comment:
Use the KeyguardManager on older devices. Only pass through this to the BiometricPrompt above version Q (= 29).